app/controllers/course/assessment/submission/answer/text_response/text_response_controller.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+class Course::Assessment::Submission::Answer::TextResponse::TextResponseController < \
+  Course::Assessment::Submission::Answer::TextResponse::Controller
+  load_resource :actable, class: Course::Assessment::Answer::TextResponse.name,
+                          singleton: true, through: :answer
+  before_action :set_text_response_answer
+
+  helper Course::Assessment::Submission::SubmissionsHelper.name.sub(/Helper$/, '')
+
+  def upload_files
+    success = @text_response_answer.class.transaction do
+      raise ActiveRecord::Rollback unless update_answer_files(upload_files_params)
+
+      true
+    end
+
+    if success
+      render @text_response_answer.answer
+    else
+      @text_response_answer.errors.messages.each do |attribute, message|
+        @text_response_answer.answer.errors.add(attribute, message)
+      end
+      render json: { errors: @text_response_answer.answer.errors.messages }, status: :bad_request
+    end
+  end
+
+  def delete_file
+    attachment_reference = AttachmentReference.find(delete_file_params[:attachment_id])
```

Comment:
Why not `@text_response_answer.attachment`?

Reply:
I understand what you mean, and it's true `text_response` answer has the concern `HasManyAttachments`.
I have changed it accordingly into `@text_response_answer.attachments` since one answer can have multiple attachments here